src/jsonpath.js

```diff
@@ -260,15 +260,31 @@ JSONPath.prototype.evaluate = function (
     if (exprList[0] === '$' && exprList.length > 1) {
         exprList.shift();
     }
+    let wantsLength = false;
+    if (exprList.at(-1) === 'length') {
```

Comment:
I think treating `.length` as length function is a bad idea, it will conflict with `length` property, if the object has it.